src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParser.java

```diff
@@ -0,0 +1,448 @@
+
+                    List<String> tableValues = getValuesList(iter);
+                    // VALUES case
+                    boolean valuesFound = false;
+                    int valuesMarker = iter.nextIndex();
+                    while (!valuesFound && iter.hasNext()) {
+                        t = iter.next();
+                        if (t.getType() == SQLServerLexer.VALUES) {
+                            valuesFound = true;
+                            do {
+                                query.getValuesList().add(getValuesList(iter));
+                            } while (iter.hasNext() && iter.next().getType() == SQLServerLexer.COMMA);
+                            iter.previous();
+                        }
+                    }
+                    if (!valuesFound) {
+                        resetIteratorIndex(iter, valuesMarker);
+                    }
+                    if (!query.getValuesList().isEmpty()) {
+                        for (List<String> ls : query.getValuesList()) {
+                            if (tableValues.isEmpty()) {
+                                query.getColumns().add("*");
+                            }
+                            for (int i = 0; i < ls.size(); i++) {
+                                if (ls.get(i).equalsIgnoreCase("?")) {
+                                    query.getColumns().add((tableValues.size() == 0) ? "?" : tableValues.get(i));
+                                }
+                            }
+                        }
+                    }
+                    break;
+                case SQLServerLexer.DELETE:
+                    t = skipTop(iter);
+                    // Check for optional keyword FROM and move the iterator back if it isn't there
+                    if (t.getType() != SQLServerLexer.FROM) {
+                        t = iter.previous();
+                    }
+                    query.getTableTarget().add(getTableTargetChunk(iter, query.getAliases(), DELETE_DELIMITING_WORDS));
+                    break;
+                case SQLServerLexer.UPDATE:
+                    skipTop(iter);
+                    t = iter.previous();
+                    // Get next chunk
+                    query.getTableTarget().add(getTableTargetChunk(iter, query.getAliases(), UPDATE_DELIMITING_WORDS));
+                    break;
+                case SQLServerLexer.FROM:
+                    query.getTableTarget().add(getTableTargetChunk(iter, query.getAliases(), FROM_DELIMITING_WORDS));
+                    break;
+                case SQLServerLexer.PARAMETER:
+                    int parameterIndex = iter.nextIndex();
+                    String columnName = SQLServerParser.findColumnAroundParameter(iter);
+                    query.getColumns().add(columnName);
+                    resetIteratorIndex(iter, parameterIndex);
+                    break;
+                default:
+                    // skip, we only support SELECT/UPDATE/INSERT/DELETE

```

Comment:
log?

Reply:
"Skipped" just means we don't act on the keyword. The word is still appended onto/used in the query.